web/ui/react-app/src/pages/Services.tsx

```diff
@@ -2,14 +2,113 @@ import React, { FC } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
 import { Alert } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { LabelsTable } from './LabelsTable';
 
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+interface DiscoveredLabels {
+  address: string;
+  metrics_path: string;
+  scheme: string;
+  job: string;
+  my: string;
+  your: string;
+}
+
+interface ActiveTargets {
+  discoveredLabels: DiscoveredLabels[];
+  labels: any;
+  scrapePool: string;
+  scrapeUrl: string;
+  lastError: string;
+  lastScrape: string;
+  lastScrapeDuration: number;
+  health: string;
+}
+
+interface ServiceMap {
+  activeTargets: ActiveTargets[];
+  droppedTargets: any[];
+}
+
+const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch<ServiceMap>(`${pathPrefix}/api/v1/targets`);
+  const processTargets = (response: ServiceMap) => {
+    console.warn(response);
+    const activeTargets = response.activeTargets;
+    const targets: any = {};
+
+    // Get targets of each type along with the total and active end points
+    for (const target of activeTargets) {
+      const { scrapePool: name } = target;
+      if (!targets[name]) {
+        targets[name] = {
+          total: 0,
+          active: 0,
+        };
+      }
+      targets[name].total++;
+      targets[name].active++;
+    }
+    for (const target of response.droppedTargets) {
+      const { scrapePool: name } = target;
+      targets[name].total++;
+    }
+
+    return targets;
+  };
+
+  const processLabels = (response: ActiveTargets[]) => {
```

Comment:
Looking at what `processTargets` and `processLabels` do, it would seem clearer to call them:

- `processTargets` -> `processSummary`
- `processLabels` -> `processTargets`